### PR TITLE
Remove PDM

### PR DESCRIPTION
**Pull request by bsweger**
We haven't made an official decision on what will replace it, but the combination of conda and pdm was causing some weirdness with virtual environments, so let's switch it out to managing dependencies via pip-tools, even if only temporarily.

### CODE REVIEW

**Comment by bsweger** [comment]
@elray1 I'm gonna start submitting PRs to the data-pipeline branch we have going. PDM was getting on my nerves, so wanted to ditch it before doing further work.

File: data-pipeline/README.md [hunk 1]
```diff
@@ -33,29 +37,63 @@ Follow the directions below to set this project up on your local machine.
...
+
+    ```bash
+    .venv\Scripts\activate
+    ```
+
+4. Install the project dependencies. The following commands can also be used to update dependencies after pulling upstream code changes:
+
+    ```bash
+    # if you're planning to run the scripts without making code changes
+    pip install -r requirements/dev-requirements.txt && pip install -e .
```
Comment (elray1) [inline data-pipeline/README.md:64]:
```suggestion
    pip install -r requirements/requirements.txt && pip install -e .
```

File: data-pipeline/README.md [hunk 2]
```diff
@@ -33,29 +37,63 @@ Follow the directions below to set this project up on your local machine.
...
+
+    # if you're planning to make and submit code changes
+    pip install -r requirements/dev-requirements.txt && pip install -e .
+    ```
+
+### Adding new dependencies
+
+This project uses [`pip-tools`](https://github.com/jazzband/pip-tools) to generate requirements files from `pyproject.toml`.
+
```
Comment (elray1) [inline data-pipeline/README.md:73]: Noting that [pip-tools docs](https://github.com/jazzband/pip-tools?tab=readme-ov-file#installation) indicate that you should install it after activating your virtual env, moving note about this here from above.

Note, I think github is eating the closing ` ``` ` in my suggestion here, can't figure out how to make it not do that...
```suggestion
This project uses [`pip-tools`](https://github.com/jazzband/pip-tools) to generate requirements files from `pyproject.toml`.
To install `pip-tools`, run the following after activating your virtual environment:

```bash
python -m pip install pip-tools
```
```

File: data-pipeline/README.md [hunk 3]
```diff
@@ -19,9 +19,13 @@ If you'd like to try it out, this section has the setup instructions.
 
 **Prerequisites**
 
-The setup instructions below use [PDM](https://pdm-project.org/) to install Python, manage a Python virtual environment, and manage dependencies. However, PDM is only absolutely necessary for managing dependencies (because the lockfile is in PDM format), so other tools for Python installs and environments will work as well.
+Before setting up the project:
+
+- Your machine will need to have an installed version of Python that meets the `requires-python` constraint in [pyproject.toml](pyproject.toml)
+- That version of Python should be set as your current Python interpreter (if you don't already have a preferred Python workflow, [pyenv](https://github.com/pyenv/pyenv) is good tool for managing Python versions on your local machine).
+
+In addition, if you're planning to make code changes that require adding or removing project dependencies, you'll need `pip-tools` installed on your machine. ([`pipx`](https://github.com/pypa/pipx) is a handy way to install python packages in a way that makes them available all the time, regardless of whatever virtual environment is currently activated.)
```
Comment (elray1) [inline data-pipeline/README.md:27]: Pip tools install instructions indicate that it should be installed within an activated virtual environment, suggestion here is to signpost that we will deal with that later in these instructions.
```suggestion
In addition, if you're planning to make code changes that require adding or removing project dependencies, you'll need `pip-tools` installed on your machine; we give installation instructions for `pip-tools` below. ([`pipx`](https://github.com/pypa/pipx) is a handy way to install python packages in a way that makes them available all the time, regardless of whatever virtual environment is currently activated.)
```

File: data-pipeline/README.md [hunk 4]
```diff
@@ -33,29 +37,63 @@ Follow the directions below to set this project up on your local machine.
...
-pdm sync
+pip-compile --extra dev -o requirements/dev-requirements.txt pyproject.toml
 ```
 
 ## Running the code
 
 Set up the project as described above and make sure the virtual environment is activated.
 
-1. From the repo's root, navigate to the directory that contains the `assign_clades.py` script::
+1. From the repo's `data-pipeline` directory, navigate to the directory that contains the `assign_clades.py` script::
```
Comment (elray1) [inline data-pipeline/README.md:93]:
```suggestion
1. From the repo's `data-pipeline` directory, navigate to the directory that contains the `assign_clades.py` script:
```

**Review by elray1 — APPROVED** [review]
Thanks for putting together these updates & detailed instructions! I was able to get it going and ran the script.  Awesome!

I made a few minor suggestions to the readme.  I'm approving because I don't feel like I need to look at it again.

File: data-pipeline/README.md (hunk 3 above)
Comment (bsweger) [reply data-pipeline/README.md:27]: I love your code reviews @elray1 -- not everyone would take the time to review pip-tools documentation as part of this 🙏 

This is an interesting point...I actually don't install `pip-tools` into my project virtual environments. I install it via `pipx`, which makes it available everywhere.

If that's one too many tools to toss around, would it be easier just to add `pip-tools` as a dev dependency here?

[I'm gonna merge this PR, so I can proceed, but can address this point in the next one]

Comment (elray1) [reply data-pipeline/README.md:27]: Adding pip-tools as a dev dependency would be helpful to me -- I definitely feel like I'm swimming in a sea of tools right now 😅. I may feel better/differently about this in a couple of months when i have workflows more sorted out for myself though.

**Pull request merged**